### PR TITLE
[SOL-2188] Make state field required for CA and US.

### PR DESCRIPTION
**Pull request by vkaracic**
https://openedx.atlassian.net/browse/SOL-2188

### CODE REVIEW

**Comment by edx-webhook** [comment]
Thanks for the pull request, @vkaracic! It looks like you're a member of a company that does contract work for edX. If you're doing this work as part of a paid contract with edX, you should talk to edX about who will review this pull request. If this work is not part of a paid contract with edX, then you should ensure that there is an OSPR issue to track this work in [JIRA](https://openedx.atlassian.net), so that we don't lose track of your pull request.

<a   href="https://openedx-webhooks.herokuapp.com/github/process_pr?repo=edx%2Fecommerce&number=1091"   rel="nofollow" >Create an OSPR issue for this pull request</a>.

**Review by mjfrey — APPROVED** [review]

File: ecommerce/static/js/pages/basket_page.js [hunk 1]
```diff
@@ -89,6 +89,10 @@ define([
                 'select[name=country]'
             ];
 
+            if ($('select[name=country').val() === 'US' || $('select[name=country').val() === 'CA') {
```
Comment (iivic) [inline ecommerce/static/js/pages/basket_page.js]: You forgot to close the parenthesis [name=country].
Just an idea `if (['US', 'CA'].indexOf($('select[name=country]').val()) > -1)`. You decide if you want to use it.

File: ecommerce/static/js/pages/basket_page.js [hunk 2]
```diff
@@ -261,11 +265,15 @@ define([
                         '<select name="state" class="select form-control" id="id_state"' +
                         'aria-required="true" required></select>'
                     );
+                    $('#id_state').append($('<option>', {value: '', text: '<Choose state/province>'}));
+                    $('#div_id_state label').text('State/Province (required)');
```
Comment (iivic) [inline ecommerce/static/js/pages/basket_page.js]: Nitpick: `$('#div_id_state label') `is a good selector but `$('#div_id_state').find('label')` is better according to https://learn.jquery.com/performance/optimize-selectors/

File: ecommerce/static/js/pages/basket_page.js [hunk 3]
```diff
@@ -261,11 +265,15 @@ define([
...
+
                     _.each(states[country], function(value, key) {
                         $('#id_state').append($('<option>', {value: value, text: key}));
                     });
                 } else {
                     $(inputDiv).empty();
+                    $('#div_id_state label').text('State/Province');
```
Comment (iivic) [inline ecommerce/static/js/pages/basket_page.js]: Nitpick: jquery selector like mentioned.

File: ecommerce/static/js/test/specs/pages/basket_page_spec.js [hunk 4]
```diff
@@ -122,8 +122,8 @@ define([
                         '<option value="CA">Canada</option>' +
                         '<option value="HR">Croatia</option>' +
                         '</select></div><p class="help-block"></p></div>' +
-                        '<div class="form-item"><div id="div_id_state"><div class="controls">' +
-                        '<input name="state"></div></div>' +
+                        '<div class="form-item"><div id="div_id_state"><label>State/Province</label>' +
+                        '<div class="controls"><input name="state"></div></div>' +
```
Comment (iivic) [inline ecommerce/static/js/test/specs/pages/basket_page_spec.js]: Not your fault but it seems that one div closing tag is missing `</div>`?

File: ecommerce/static/js/test/specs/pages/basket_page_spec.js [hunk 5]
```diff
@@ -136,6 +136,7 @@ define([
 
                     $('select[name=country]').val('CA').trigger('change');
                     expect($('#id_state').prop('tagName')).toEqual('SELECT');
+                    expect($('#div_id_state label').text()).toEqual('State/Province (required)');
```
Comment (iivic) [inline ecommerce/static/js/test/specs/pages/basket_page_spec.js]: Nitpick: jquery selector like mentioned.

**Review by iivic — CHANGES_REQUESTED** [review]

File: ecommerce/static/js/pages/basket_page.js (hunk 2 above)
Comment (vkaracic) [reply ecommerce/static/js/pages/basket_page.js]: I'm not sure if there is an increase in speed.
```
although the speed improvement may be diminished by the subsequent call to .find().
```

**Comment by vkaracic** [comment]
@iivic comments addressed, please take another look.

**Review by iivic — APPROVED** [review]

**Pull request merged**